domain/src/test/java/org/oppia/domain/util/StateRetrieverTest.kt

```diff
@@ -235,6 +237,90 @@ class StateRetrieverTest {
     assertThat(ruleSpecMap.inputMap["x"]).isEqualTo(imageRegionSelectionIsInRegionValue)
   }
 
+  @Test
+  fun testParseState_withRatioInputInteraction_parsesRuleEqualsRuleSpec() {
+    val state = createStateFromJson(
+      "RatioExpressionInput",
+      RATIO_TEST_EXPLORATION_NAME
+    )
+    val ruleSpecMap = state.interaction.answerGroupsList
+      .flatMap(AnswerGroup::getRuleSpecsList)
+      .associateBy(RuleSpec::getRuleType)
+    assertThat(ruleSpecMap).containsKey("Equals")
+  }
+
+  @Test
+  fun testParseState_withImageRegionSelectionInteraction_parsesRuleWithEqualsWithValueAtX() {
+    val state = createStateFromJson(
+      "RatioExpressionInput",
+      RATIO_TEST_EXPLORATION_NAME
+    )
+    val ruleSpecMap = lookUpRuleSpec(state, "Equals")
+
+    val ratioList = listOf(4, 5)
+
+    val ratioInputEqualsValue =
+      InteractionObject.newBuilder().setRatioExpression(createRatio(ratioList)).build()
+
+    assertThat(ruleSpecMap.inputMap["x"]).isEqualTo(ratioInputEqualsValue)
```

Comment:
```suggestion
    val expectedInputRatio = createRatio(listOf(4, 5))
    val expectedInputInteractionObject =
      InteractionObject.newBuilder().setRatioExpression(expectedInputRatio).build()
    assertThat(ruleSpecMap.inputMap["x"]).isEqualTo(expectedInputInteractionObject)
```

Some restructuring suggestions to make things a bit clearer (renames + ensuring the 'assertion' step is together).

Ditto below for the other new tests.

Reply:
done